test/examples/util/util.go

```diff
@@ -0,0 +1,174 @@
+		Command: cmd,
+		Timeout: CmdTimeout,
+		Env:     environment,
+	}
+	fmt.Printf("=> Command to execute: %v \n", currentCmd)
+	if workingDirOp != nil {
+		return icmd.RunCmd(currentCmd, workingDirOp)
+	}
+	return icmd.RunCmd(currentCmd)
+}
+
+//SetDir sets the working directory
+func SetDir(path string) {
+	workingDirOp = icmd.Dir(path)
+}
+
+//SetKubeConfig sets the KUBECONFIG to the cluster
+func SetKubeConfig(kc string) {
+	kubeConfig = kc
+}
+
+//GetExamplesDir returns a root path to the examples
+func GetExamplesDir() string {
+	wd, err := os.Getwd()
+	if err != nil {
+		log.Fatalf("Failed to get the working dir.")
+	}
+	return path.Clean(fmt.Sprintf("%s/../../examples", wd))
+}
+
+//GetOutput returns the output using Stdout()
+func GetOutput(res *icmd.Result) string {
+	var output string
+	exitCode := res.ExitCode
+	if exitCode == 0 || exitCode == 127 {
+		output = res.Stdout()
+	} else {
+		output = res.Stderr()
+	}
+	fmt.Printf("OUTPUT: %s \n", output)
+	return output
+}
+
+//GetProjectCreationResult returns the result of make create-project
+func GetProjectCreationResult(projectRes string, project string) string {
+	item := ""
+	lstArr := strings.Split(projectRes, "\n")
+	for _, item := range lstArr {
+		if strings.Contains(item, project) {
+			fmt.Printf("item matched as %s \n", item)
+			return item
+		}
+	}
+	return item
+}
+
+//GetCmdResult executes execCmd function indefinitely till the default timeout occurs if there is no response of a command, returns the result(res) immediately
+func GetCmdResult(status string, item ...string) string {
+	return GetCmdResultWithTimeout(status, defaultRetryInterval, defaultRetryTimeout, item...)
+}
+
+//GetCmdResultWithTimeout executes execCmd function indefinitely till the timeout occurs if there is no response of a command, returns the result(res) immediately
+func GetCmdResultWithTimeout(status string, retryInterval time.Duration, retryTimeout time.Duration, item ...string) string {
+	var res string
+	cntr = 0
+	checkFlag := false
+	err := wait.PollImmediate(retryInterval, retryTimeout, func() (bool, error) {
+		checkFlag, res = execCmd(item...)
+		if checkFlag {
+			if status != "" {
+				if strings.Contains(res, status) {
+					return true, nil
+				}
+				return false, nil
+			}
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		return ""
+	}
+	return res
+}
+
+//execCmd returns a boolean result and the result of the command (cmdRes) executed
+func execCmd(item ...string) (bool, string) {
+	cntr++
+	var cmdRes string
+	checkFlag := false
+	fmt.Printf("Get result of the command...iteration %v \n", cntr)
+	cmdRes = GetOutput(Run(item...))
+	if cmdRes != "" {
+		checkFlag = true
+	}
+	return checkFlag, cmdRes
+}
+
+//GetPodLst fetches the list of pods
+func GetPodLst(operatorsNS string) string {
+	log.Print("Fetching the list of running pods")
+	pods := GetCmdResult("", "oc", "get", "pods", "-n", operatorsNS, "-o", `jsonpath={.items[*].metadata.name}`)
```

Comment:
Instead of running direct command should we be using [k8s client-go](https://github.com/kubernetes/client-go) or [openshift client-go](https://github.com/openshift/client-go). This might give us better control over cluster side error and output handling and we can also use k8s friendly types like [schema.GroupVersionResource](https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#GroupVersionResource). Then this helper or utils pkg also might be useful to use in e2e tests if needed. WDYT does it make sense?

Reply:
> Instead of running direct command should we be using [k8s client-go](https://github.com/kubernetes/client-go) or [openshift client-go](https://github.com/openshift/client-go). This might give us better control over cluster side error and output handling and we can also use k8s friendly types like [schema.GroupVersionResource](https://godoc.org/k8s.io/apimachinery/pkg/runtime/schema#GroupVersionResource). Then this helper or utils pkg also might be useful to use in e2e tests if needed. WDYT does it make sense?

@pratikjagrut Thanks for the opinion. However, the idea here is to simulate real user's experience end-to-end where the user works with the OpenShift using the `oc`.

Reply:
@pmacik Okay.